src/components/ProfilePage.tsx

```diff
@@ -0,0 +1,221 @@
+                userProfile.summary = summary as string
+                userProfile.url = url as string
+                userProfile.profilePicureFileHandleId = fileHandleId as string
+                await updateMyUserProfile(userProfile, accessToken)
+                displayToast('Profile has been successfully updated', 'success')
+                getProfile()
+                setEditing(false)
+            }
+        } catch(err:any){
+            displayToast(err.reason as string, 'danger')
+        }
+    }
+
+    useEffect(()=>{
+        getProfile()
+    },[accessToken,])
+
+    const EditField = (label: string, updatedValue: string|undefined, updateFn: Function) => {
+        return(
+            <div className='edit-field'>
+                <FormLabel>{label}</FormLabel>
+                <FormControl
+                    onChange={e=>updateFn(e.target.value)}
+                    value={updatedValue}
+                    placeholder= {`Enter ${label}`}
+                /> 
+            </div>
+        )
+    }
+
+    const onCancel = async () => {
+        getProfile()
+        setEditing(false)
+    }
+
+    const uploadHandler = async(e: React.ChangeEvent<HTMLInputElement>) => {
+        if(e.target.files){
+            const file = e.target.files[0]
+            try{
+                const resp: FileUploadComplete = await uploadFile(accessToken, file.name, file)
+                const picUrl = await getFileHandleByIdURL(resp.fileHandleId,accessToken)
+                setFileHandleId(resp.fileHandleId)
+                setProfilePicUrl(picUrl)
+            } catch(err:any){
+                console.error(err)
+            }
+        }
+    }
+
+    const hiddenFileInput = React.useRef<HTMLInputElement>(null)
+    const clickHandler = () => {
+        if (hiddenFileInput?.current!) {
+          hiddenFileInput.current?.click()
+        }
+    }
+
+    const uploadImg = (
+        <>
+            <input
+                type={'file'}
+                ref={hiddenFileInput}
+                onChange={uploadHandler}
+                style={{display: 'none'}}
+            />
+            <Button style={{marginTop:'16px'}} variant='secondary' onClick={clickHandler}>Upload File</Button>
+        </>
+    )
+
+    const validProfilePic = (
+        <>
+            {verified ? <div className='verified-img-container'>
+                <img className='verified-border' src={VerifiedBorder}/>
+                <img className='verified-img' src={profilePicUrl}/>
+                {editing && uploadImg}
+            </div> :
+            <div className='non-verified-profile-pic'>
+                <img className='non-verified-img' src={profilePicUrl}/>
+                {editing && uploadImg}
+            </div>
+            }
+        </>
+    ) 
+
+    return(
+        <div className="bootstrap-4-backport blue-background profile-page">
+            <Container>
+                <div className='edit-btn-container'>
+                    {!!!editing ? 
```

Comment:
Please change all of these triple exclamation marks to single (removing the extra double negation).  
I think _!editing_ is already a boolean.